helper/cert/source_gen.go

```diff
@@ -51,11 +67,26 @@ func (s *GenSource) Certificate(ctx context.Context, last *Bundle) (Bundle, erro
 	defer s.mu.Unlock()
 	var result Bundle
 
+	if s.LeaderElector != nil {
+		leaderCheck, err := s.LeaderElector.IsLeader()
+		if err != nil {
+			return result, err
+		}
+		// For followers, run different function here that reads bundle from Secret,
+		// and returns that in the result. That will flow through the existing
+		// notify channel structure, testing if it's the same cert as last, etc.
+		if !leaderCheck {
```

Comment:
Can we get a log saying whether this pod is the leader or follower? Also wondering if we should apply a label so we can query for the leader (not blocking, future enhancement probably).

Currently I can't tell which is which:

```
[~/Git/vault-agent-demo] kubectl logs vault-injector-57485f744-24jwd -c sidecar-injector
Using leader elector logic
2020-12-16T15:22:43.623Z [INFO]  handler: Starting handler..
Listening on ":8080"...
Updated certificate bundle received. Updating certs...
[~/Git/vault-agent-demo] kubectl logs vault-injector-57485f744-lts8w -c sidecar-injector
Using leader elector logic
2020-12-16T15:22:44.109Z [INFO]  handler: Starting handler..
Listening on ":8080"...
Updated certificate bundle received. Updating certs...
```

Reply:
Yep, good point, added more logging in ac4b4f4. I set the leader and follower log lines to debug since they are logged pretty frequently.